### ISSUE TITLE
training examples

### ISSUE THREAD

**Issue by bghira** [Stability-AI/sd3.5#5]
we need internal training loop loss function, the same code used to do training time predictions

all i see is Lykon talking about how stable sd3.5L is but i dont see this bearing out in reality

think of my position as maintainer of simpletuner who wasnt really consulted at any point of the sd35 finetuning tutorial's creation.

the tutorial itself outlines the damage that occurred to the model when training. i find it hard to believe that this is normal results or mirrors in any way what was seen during pretraining.

it feels just like finetuning Flux.1 Schnell, but SD3L isnt even distilled, right?

anyway please provide the code examples needed or it feels like it is necessry for me to remove sd3.x support from simpletuner moving forward as it is not a good user experience to waste money.

**Comment by bghira** [comment]
learning rate 1e-5 for lycoris does nothing and then kaboom.. but flux trains fine.

![image0-19.png](https://github.com/user-attachments/assets/4d74c3da-96a4-45e6-8b39-6e2b92e88b37)

**Comment by DriveHabits** [comment]
@bghira seems like a skill issue

**Comment by bghira** [comment]
if they provide the training example script then we can definitively state this one way or another. i would prefer it be a "skill issue".

**Comment by bghira** [comment]
well as a part of my contribution to the issue, i was using Flux VAE outputs by accident to train the model instead of SD3 VAE outputs. embarrassing, yes. but still, the lack of canonical training example leaves too much to the imagination when searching down issues and would still love to have access to one for reference.

**Comment by DarkViewAI** [comment]
> well as a part of my contribution to the issue, i was using Flux VAE outputs by accident to train the model instead of SD3 VAE outputs. embarrassing, yes. but still, the lack of canonical training example leaves too much to the imagination when searching down issues and would still love to have access to one for reference.
> 
> updated test result;
> 
> ![image](https://private-user-images.githubusercontent.com/59658056/379452843-5c03ee89-7e0d-40da-83ab-f90fe3fbf0fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3MTY4NDgsIm5iZiI6MTcyOTcxNjU0OCwicGF0aCI6Ii81OTY1ODA1Ni8zNzk0NTI4NDMtNWMwM2VlODktN2UwZC00MGRhLTgzYWItZjkwZmUzZmJmMGZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIzVDIwNDkwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTViMmFhY2NkMjQ1ZTc1Mjk0YTU2MGRlNDMyMWI4MDM2ZTIwNTEyZjU3YTdhYThhNTRlODdiMzE2YTllMDkyOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JEjlH9tQoU-CkOsEgD8gccpwVenMWLVCnGH7Wx6DSmU)

as the guy said above, skill issue. P.s you seem very toxic and senile. You need to grow up bud, you constantly shitting on kohya and onetrainer is wild considering you've taken so much code from them. 

do better.

**Comment by DarkViewAI** [comment]
love how you were shit talking lykon, but you were the one to make the mistake lol

**Comment by bghira** [comment]
i don't see your contributions anywhere..

**Comment by bghira** [comment]
this is from the official fine-tuning tutorial as published by StabilityAI

![image](https://github.com/user-attachments/assets/b8fdb9b8-535a-4e9c-8342-d5d4dd0ddbf4)

**Comment by bghira** [comment]
the tutorial is [here](https://stabilityai.notion.site/Stable-Diffusion-3-5-Large-Fine-tuning-Tutorial-11a61cdcd1968027a15bdbd7c40be8c6); my trainer, simpletuner, was used for this tutorial because it gives the best results. 

it gives the best results because i work hard on the implementations. i do deep dives, and investigate, and ask for help when i'm not seeing the expected results.

i want anyone to know that making mistakes and talking about them is fine, as it helps others who come along later with the same struggles.

accusing others of code theft or being "toxic" while simultaneously calling them senile is just kinda unwarranted, doesn't help push the discussion forward. if you think something has been stolen, try raising it as an issue on the bug tracker for either my project or kohya's or onetrainer, so that we can ensure proper attribution. however, it should be noted that i rely on Diffusers and have no need to steal code from anybody else. furthermore, simpletuner produces the best results and is updated to support new features on a more rapid pace than kohya's or nerogar's projects. those two are great developers, and i admire their tenacity and skill. i have no beef or issue with them, quite the opposite.

cc @brianfitzgerald @voletiv

**Comment by AmericanPresidentJimmyCarter** [comment]
Teaching machines to learn can make anyone grumpy, especially when you try to teach them nonsense and expect something else 😁

**Comment by voletiv** [comment]
> well as a part of my contribution to the issue, i was using Flux VAE outputs by accident to train the model instead of SD3 VAE outputs. embarrassing, yes. but still, the lack of canonical training example leaves too much to the imagination when searching down issues and would still love to have access to one for reference.
> 
> updated test result;
> 
> ![image](https://private-user-images.githubusercontent.com/59658056/379452843-5c03ee89-7e0d-40da-83ab-f90fe3fbf0fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3OTYyNjUsIm5iZiI6MTcyOTc5NTk2NSwicGF0aCI6Ii81OTY1ODA1Ni8zNzk0NTI4NDMtNWMwM2VlODktN2UwZC00MGRhLTgzYWItZjkwZmUzZmJmMGZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI0VDE4NTI0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2NWFjODhlNWYyMzc1YzIwNWQ3ZDBjNTZiZWEwNTJhODM2YjQyMjUxM2EwNDFiMmViMGI2NWRmYmRjZDI5ZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XAeuCGJjBJ6chSBqS6HSPS12D6ky4xTTmCLEOxGyBjo)

Cool! I guess it's training fine now, I'm closing this issue.
